gnucash/report/report-system/html-document.scm

```diff
@@ -128,7 +128,8 @@
 ;; returns the html document as a string, I think.
 (define (gnc:html-document-render doc . rest)
   (let ((stylesheet (gnc:html-document-style-sheet doc))
-        (headers? (if (null? rest) #f (if (car rest) #t #f)))
+        (headers? (if (null? rest) #t (if (not (car rest)) #f #t)))
```

Comment:
Did you really mean to flip the first #f to #t?

Reply:
yes. if no further options are given, then the default behaviour should be to create the header. I think that is how it was before. possibly I made it #f myself for testing and then forgot to make it #t again...

Reply:
No, it was #f. The only place gnc:html-document-render-data is called with only one arg is in gnc:options->render and that isn't used in a way that the presence or absence of headers matters so it's really moot what the default is, but it's also not germane to your change, but changing the default of a function deserves a separate commit of its own with an explanation of why the old default is wrong, and it shouldn't be buried in a PR that's ostensibly about adding tests.

The second part of the change is non-functional: ```(if (car rest) #t #f)``` and ```(if (not (car rest)) #f #t)``` are the same thing.

Reply:
yes, no, and yes

yes
it was #f before, so my statement above is incorrect, I flipped it by intention

no
and I do not vote for reverting back to 
reason:
because of  Bug796826 - Report HTML - not possible to suppress the document header
the header was always present, i.e. no matter if args are given or not, the default behaviour was to always include the header. If we leave it like that, chances are that it'll break user's custom reports.
Please confirm to revert back to #f.

yes
The second part of the change is non-functional and should be reverted back.

Reply:
Ah, got it. The default here didn't matter because of the error on line 152. OK.